src/super_gradients/training/models/detection_models/customizable_detector.py

```diff
@@ -89,7 +107,71 @@ def __init__(
         self._default_multi_label_per_box = True
         self._default_class_agnostic_nms = False
 
+    def forward_sliding_window(self, images):
+        batch_size, _, _, _ = images.shape
+        all_detections = [[] for _ in range(batch_size)]  # Create a list for each image in the batch
+
+        # Generate and process each tile
+        for img_idx in range(batch_size):
+            single_image = images[img_idx : img_idx + 1]  # Extract each image
+            tiles = self._generate_tiles(single_image, self.tile_size, self.tile_step)
+            for tile, (start_x, start_y) in tiles:
+                tile_detections = self.forward_whole_image(tile)
+                # Apply local NMS using post_prediction_callback
+                tile_detections = self.sliding_window_post_prediction_callback(tile_detections)
+                # Adjust detections to global image coordinates
+                for img_i_tile_detections in tile_detections:
+                    if len(img_i_tile_detections) > 0:
+                        img_i_tile_detections[:, :4] += torch.tensor([start_x, start_y, start_x, start_y], device=tile.device)
+                        all_detections[img_idx].append(img_i_tile_detections)
+
+        # Concatenate and apply global NMS for each image's detections
+        final_detections = []
+        for detections in all_detections:
+            if detections:
+                detections = torch.cat(detections, dim=0)
+                # Apply global NMS
+                pred_bboxes = detections[:, :4]
+                pred_cls_conf = detections[:, 4]
+                pred_cls_label = detections[:, 5]
+                idx_to_keep = torchvision.ops.boxes.batched_nms(
+                    boxes=pred_bboxes, scores=pred_cls_conf, idxs=pred_cls_label, iou_threshold=self._default_nms_iou
+                )
+
+                final_detections.append(detections[idx_to_keep])
+            else:
+                final_detections.append(torch.empty(0, 6).to(images.device))  # Empty tensor for images with no detections
+        return final_detections
+
+    def _generate_tiles(self, image, tile_size, tile_step):
+        _, _, h, w = image.shape
+        tiles = []
+
+        # Calculate the end points for the grid
+        max_y = h if (h - tile_size) % tile_step < self.min_tile_threshold else h - (h - tile_size) % tile_step + tile_size
```

Comment:
Ok, I'm already lot in this logic. I probably can grasp what is going on here if I take a sheet of paper and a pen. But is there a specific reason we want to do it this way and not like so:

```
def _generate_tiles(self, image, tile_size, tile_step):
  image_height, image_width = image.shape[:2]
  start_y = 0
  while start_y > image_height:
     start_x = 0
     while start_x > image_width:
       crop = image[start_y: start_y+tile_size, start_x:start_x+tile_size]
       yield crop, (start_y, start_x)
       start_x += tile_step
  start_y += tile_step

```

Reply:
I think you wouldv'e got pretty easily if I added docs (: 
@ofrimasad suggested we add a min_tile_threshold parameter, so that for example in the case where:
image size = 640,641
tile_size = 640
tile_step = 640
It would completely irrleveant to pass a 1X640 padded to 640 in the second tile, so just skip it.

So the first 2 lines just say:
"calculate the end points of the tile for padding, if the remainder when dividing by the tile size is smaller then min_tile_threshold -> leave that tile out"